Adjust skip on empty (#288)
yiisoft/validator: README.md

#### Skipping empty values

By default, empty values are validated. That is undesirable if you need to allow not specifying a field.
To change this behavior, use `skipOnEmpty: true`:

```php
use Yiisoft\Validator\Rule\Number;

new Number(asInteger: true, max: 100, skipOnEmpty: true);
```

What exactly to consider to be empty is vague and can vary depending on a scope of usage.

`skipOnEmpty` is more like a shortcut, `skipOnEmptyCallback` argument checks if a given value is empty:

- If `skipOnEmpty` is `false`, `Yiisoft\Validator\SkipOnEmptyCallback\SkipNever` is used automatically for 
`skipOnEmptyCallback` - every value is considered non-empty and validated without skipping (default).
- If `skipOnEmpty` is `true`, `Yiisoft\Validator\SkipOnEmptyCallback\SkipOnEmpty` is used automatically for

> `skipOnEmptyCallback` - only non-empty values (`null`, `[]`. `''`) are validated.

Reviewer: ```suggestion
`skipOnEmptyCallback` - only non-empty values (not `null`, `[]`, or `''`) are validated.
```

- If `skipOnEmptyCallback` is set, it takes precedence to determine emptiness.

> While using first option is usually enough for HTML forms, the second one is more suitable for APIs.

Reviewer: ```suggestion
Using first option is usually good for HTML forms. The second one is more suitable for APIs.
```

The empty values can be also limited to `null` only:

```php
use Yiisoft\Validator\Rule\Number;
use Yiisoft\Validator\SkipOnEmptyCallback\SkipOnNull;

new Number(asInteger: true, max: 100, skipOnEmptyCallback: new SkipOnNull());
```

> Note that in this case `skipOnEmpty` will be automatically set to `true`, so no need to do it manually.

Reviewer: ```suggestion
Note that in this case `skipOnEmpty` will be automatically set to `true`, so there is no need to do it manually.
```

> For even more customization you can use own class implementing `__invoke()` magic method:

Reviewer: ```suggestion
For even more customization you can use your own class implementing `__invoke()` magic method:
```
